aialphanovatide/ai-alpha-api-tests
# Test/news bots coins endpoint

This pull request introduces a comprehensive test suite for the News Bot API endpoints. Below is a detailed breakdown of the additions:

Changes Introduced:

**GET Endpoints:**
Added tests for retrieving all bots (test_get_bots).
Added tests for retrieving metrics for a specific bot (test_get_news_bot_metrics).
Added tests for retrieving details of a specific bot by ID (test_get_bot).
**POST Endpoints:**
Added tests for creating a new bot with associated data (test_post_bot).
**PUT Endpoints:**
Added tests for updating an existing bot (test_put_news_bot).
**DELETE Endpoints:**
Added tests for deleting a bot (test_delete_bot).

**Key Features:**
_Parameterized Testing:_
Use of pytest.mark.parametrize for data-driven testing, improving coverage across multiple test cases.
_Schema Validation:_
Response validation against predefined schemas (NEWS_BOTS_SCHEMA, POST_NEWS_BOT_RESPONSE, etc.) using the Assertions.validate_schema method.
_Dynamic Test Data Handling:_
Utilization of Handlers.create_test_tuple for dynamically generating test cases from JSON files.
_Environment Variable Integration:_
Base URL and configurations are retrieved from environment variables using dotenv.
_Cleanup Logic:_
Automated cleanup of test-created entities (e.g., deleting a bot after a successful POST).
_Files Added/Modified:_
Test File: Added test_news_bots.py containing all test cases.
Schema Files: Updated test_data/schemas.py with additional schemas for response validation.
Test Data: Added JSON files (post_news_bot.json, put_news_bot.json, etc.) for parameterized tests.

How to Test:
Ensure all dependencies are installed (pytest, python-dotenv, etc.).
Configure the .env file with the required environment variables (NEWS_BOT_URL).
Run the test suite:
```bash
pytest test_news_bots.py
```
Verify that all tests pass and outputs match expected results.
![image](https://github.com/user-attachments/assets/1568e8c8-88ff-468f-816c-0e4b7097e3e5)
